chitchat/src/state.rs

```diff
@@ -50,11 +61,11 @@ impl NodeState {
     ) -> impl Iterator<Item = (&str, &VersionedValue)> {
         self.key_values
             .iter()
-            .filter(move |&(key, versioned_value)| predicate(key, versioned_value))
+            .filter(move |(key, versioned_value)| predicate(key, versioned_value))
             .map(|(key, record)| (key.as_str(), record))
     }
 
-    /// Returns an iterator over the version values that are older than `floor_version`.
+    /// Returns an iterator over the version values that are strictly older than `floor_version`.
```

Comment:
I would use `greater` instead of `old`. Greater version are in fact more "recent".